src/geos_types.jl

```diff
@@ -106,12 +155,29 @@ end
 
 mutable struct MultiPolygon <: AbstractGeometry
     ptr::GEOSGeom
-
+    # create multipolygon using a multipolygon or polygon pointer, else error
     function MultiPolygon(ptr::GEOSGeom)
-        multipolygon = new(ptr)
+        id = LibGEOS.geomTypeId(ptr)
+        if id == GEOS_MULTIPOLYGON
+            multipolygon = new(cloneGeom(ptr))
+        elseif id == GEOS_POLYGON
+            multipolygon = new(createCollection(
+                                GEOS_MULTIPOLYGON,
+                                GEOSGeom[cloneGeom(ptr)]))
+        else
+            error("LibGEOS: Can't convert a pointer to an element with a GeomType ID of $id to a multi-polygon (yet). Please open an issue if you think this conversion makes sense.")    
+        end
         finalizer(destroyGeom, multipolygon)
         multipolygon
     end
+
+    # create multipolygon from list of Polygon objects
+    MultiPolygon(polygons::Vector{Polygon}) = MultiPolygon(
+        createCollection(
+            GEOS_MULTIPOLYGON,
+            GEOSGeom[poly.ptr for poly in polygons],))
```

Comment:
I am not sure. My thought process was that this local collection takes ownership of the polygons, but then the call to MultiPolygon clones the collection. Then, the local collection is out of scope. Would this "return" the ownership of the polygons? I was trying to avoid cloning each of the polygons and then cloning the whole collection. But I agree, I was unsure of it my logic was correct. Thoughts?

Reply:
Oh you are right, I forgot we clone inside the call to `MultiPolygon(...)`. It might be helpful to leave a comment here to that effect, but I'll resolve this comment thread either way, thanks for explaining!